luma/core/bitmap_font.py

```
@@ -0,0 +1,632 @@
+        """
+        Utility function to create an efficient serializable representation
+        of a luma.core.bitmap_font
+        """
+        cell_size = (self.width, self.height)
+
+        area = self.count * self.width * self.height
+        table_width = ((ceil(area**0.5) + self.width - 1) // self.width) * self.width
+        table_height = ((ceil(area / table_width) + self.height - 1) // self.height) * self.height
+
+        image = Image.new('1', (table_width, table_height))
+        metrics = []
+        for i, v in enumerate(self.metrics):
+            offset = i * cell_size[0]
+            left = offset % table_width
+            top = (offset // table_width) * cell_size[1]
+            image.paste(v['img'], (left, top))
+            metrics.append((v['xwidth'], v['dst']))
+
+        fontdata = {}
+        fontdata['type'] = 'LUMA.CORE.BITMAP_FONT'
+        fontdata['count'] = len(self.metrics)
+        if self.regular:
+            fontdata['xwidth'] = self.regular[0]
+            fontdata['glyph_size'] = (self.width, self.height)
+        fontdata['cell_size'] = cell_size
+        fontdata['mappings'] = self.mappings
+        fontdata['sprite_table_dimensions'] = (table_width, table_height)
+        fontdata['sprite_table'] = image.tobytes()
+        if not self.regular:
+            fontdata['metrics'] = metrics
+
+        return fontdata
+
+    def _load_fontdata(self, fontdata):
+        """
+        Initialize font from deserialized data
+        """
+        try:
+            count = fontdata['count']
+            xwidth = fontdata.get('xwidth')
+            glyph_size = fontdata.get('glyph_size')
+            cell_size = fontdata['cell_size']
+            self.mappings = fontdata['mappings']
+            table_width, table_height = fontdata['sprite_table_dimensions']
+            image = Image.frombytes('1', (table_width, table_height), fontdata['sprite_table'])
+            metrics = fontdata.get('metrics')
+        except Exception:
```

Is there anything more relevant to catch here besides the absolute base `Exception`? This will swallow everything.

---

Now catching more specific exceptions.